okhttp/src/main/java/okhttp3/internal/io/FileSystem.kt

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.internal.io
+
+import okio.Source
+import okio.source
+import okio.Sink
+import okio.sink
+import okio.appendingSink
+import java.io.File
+import java.io.FileNotFoundException
+import java.io.IOException
+
+/**
+ * Access to read and write files on a hierarchical data store. Most callers should use the
+ * [SYSTEM] implementation, which uses the host machine's local file system. Alternate
+ * implementations may be used to inject faults (for testing) or to transform stored data (to add
+ * encryption, for example).
+ *
+ * All operations on a file system are racy. For example, guarding a call to [source] with
+ * [exists] does not guarantee that [FileNotFoundException] will not be thrown. The
+ * file may be moved between the two calls!
+ *
+ * This interface is less ambitious than [java.nio.file.FileSystem] introduced in Java 7.
+ * It lacks important features like file watching, metadata, permissions, and disk space
+ * information. In exchange for these limitations, this interface is easier to implement and works
+ * on all versions of Java and Android.
+ */
+interface FileSystem {
+
+  companion object {
+    /** The host machine's local file system.  */
+    @JvmField
+    val SYSTEM: FileSystem = object : FileSystem {
+      @Throws(FileNotFoundException::class)
+      override fun source(file: File): Source {
+        return file.source()
+      }
+
+      @Throws(FileNotFoundException::class)
+      override fun sink(file: File): Sink {
+        try {
+          return file.sink()
+        } catch (e: FileNotFoundException) {
+          // Maybe the parent directory doesn't exist? Try creating it first.
+          file.parentFile.mkdirs()
+          return file.sink()
+        }
+
+      }
+
+      @Throws(FileNotFoundException::class)
+      override fun appendingSink(file: File): Sink {
+        try {
+          return file.appendingSink()
+        } catch (e: FileNotFoundException) {
```

Comment:
nit: `_: FileNotFoundException`

Reply:
You mean space between the `e` and the `:`?

Reply:
No, since `e` is not used inside the `catch` body you can rename it to `_`.

Reply:
7024be2b154951432f4b79131441b007674a19fe